modules/osm/tags.js

```diff
@@ -13,6 +13,33 @@ export function osmSetAreaKeys(value) {
     osmAreaKeys = value;
 }
 
+export const osmLifecyclePrefixes = {
+    // nonexistent, might be built
+    proposed: true, planned: true,
+    // under maintentance or between groundbreaking and opening
+    construction: true,
+    // existent but not functional
+    disused: true,
+    // dilapidated to nonexistent
+    abandoned: true, was: true,
```

Comment:
There’s also [`not:`](https://wiki.openstreetmap.org/wiki/Key:not:), which is mostly used for `not:name` but could also be used on feature keys too.

Reply:
does it make sense to include `not:` in this list though? 

Considering a closed way tagged with `barrier=wall` + `not:amenity=school`:

- It should be treated as a line, not an area (?)
- It should not be included in the past/future filter
- It should not be rendered with the dashed line like abandoned/disused features.

Reply:
You’re right, I was thinking of the case of something that isn’t a building but isn’t anything else mappable either, such as [this patch of pavement](https://www.openstreetmap.org/way/658082823) that isn’t a building. It’s probably rare enough not to worry about yet.